dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/localfiletreetable/LocalFileTreeTablePresenter.java

```diff
@@ -474,36 +495,39 @@ private void startPutJob(final Session session,
             final JobInterruptionStore jobInterruptionStore,
             final TreeItem<Ds3TreeTableValue> remoteDestination) {
         final Ds3Client client = session.getClient();
-        final Ds3PutJob putJob = ds3PutJobFactory.createDs3PutJob(files, bucket, targetDir, remoteDestination);
-        putJob.setOnSucceeded(SafeHandler.logHandle(event -> {
-            LOG.info("BULK_PUT job {} Succeed.", putJob.getJobId());
+
+        final ImmutableList.Builder<kotlin.Pair<String, Path>> builder = ImmutableList.builder();
+        files.forEach(f -> builder.add(new kotlin.Pair<>(f.getKey(), f.getValue())));
+
+        final PutJob putJob = new PutJob(new PutJobData(builder.build(), targetDir, bucket, new JobTaskElement(settingsStore, loggingService, dateTimeUtils, client, jobInterruptionStore, savedJobPrioritiesStore)));
+        final JobTask jobTask = new JobTask(putJob);
+        jobTask.setOnSucceeded(SafeHandler.logHandle(event -> {
+            LOG.info("BULK_PUT job {} Succeed.", putJob.jobUUID());
+
             refreshBlackPearlSideItem(remoteDestination);
         }));
-        putJob.setOnFailed(SafeHandler.logHandle(failEvent -> {
+        jobTask.setOnFailed(SafeHandler.logHandle(failEvent -> {
             final Throwable throwable = failEvent.getSource().getException();
             LOG.error("Put Job Failed", throwable);
             loggingService.logMessage("Put Job Failed with message: " + throwable.getClass().getName() + ": " + throwable.getMessage(), LogType.ERROR);
+            refreshBlackPearlSideItem(remoteDestination);
         }));
-        putJob.setOnCancelled(SafeHandler.logHandle(cancelEvent -> {
-            final UUID jobId = putJob.getJobId();
-            if (jobId != null) {
-                try {
-                    client.cancelJobSpectraS3(new CancelJobSpectraS3Request(putJob.getJobId()));
-                } catch (final IOException e) {
-                    LOG.error("Failed to cancel job", e);
-                }
-                LOG.info("BULK_PUT job {} Cancelled.", jobId);
-                loggingService.logMessage(resourceBundle.getString("putJobCancelled"), LogType.SUCCESS);
+        jobTask.setOnCancelled(SafeHandler.logHandle(cancelEvent -> {
+            final UUID jobId = putJob.jobUUID();
+            try {
+                client.cancelJobSpectraS3(new CancelJobSpectraS3Request(putJob.jobUUID()));
+            } catch (final IOException e) {
+                LOG.error("Failed to cancel job", e);
```

Comment:
Should we log something to the gui in the event that this condition happens?

Reply:
This has been corrected however it didn't close the issue, see line 516 of current file.